install-rust-toolchain.sh

```diff
@@ -158,17 +158,12 @@ function install_rust_toolchain() {
 }
 
 function source_cargo() {
-    if [[ -e "${HOME}/.cargo/env" ]]; then
-        source "${HOME}/.cargo/env"
-        export CARGO_HOME="${HOME}/.cargo"
+    if [[ -e "${CARGO_HOME}/env" ]]; then
+        source "${CARGO_HOME}/env"
     else
-        if [[ -n "${CARGO_HOME}" ]] && [[ -e "${CARGO_HOME}/env" ]]; then
-            source ${CARGO_HOME}/env
-        else
-            echo "Warning: Unable to source .cargo/env"
-            export CARGO_HOME="${HOME}/.cargo"
-        fi
+        echo "Warning: Unable to source ${CARGO_HOME}/env"
```

Comment:
```suggestion
        echo "Warning: Unable to source ${CARGO_HOME}/env. Verify that the path specified for ${CARGO_HOME} exists"
```

I would suggest adding a hint, the user can specify a wrong CARGO_HOME